astropy/units/core.py

```diff
@@ -1142,14 +1147,14 @@ def compose(self, equivalencies=[], units=None, max_depth=2,
 
         Parameters
         ----------
-        equivalencies : list of equivalence pairs, optional
+        equivalencies : list of (2,) tuple
             A list of equivalence pairs to also list.  See
             :ref:`unit_equivalencies`.
             This list is in addition to possible global defaults set by, e.g.,
             `set_enabled_equivalencies`.
             Use `None` to turn off all equivalencies.
 
-        units : set of units to compose to, optional
+        units : set of :class:`~astropy.units.Unit` to compose to, optional
```

Comment:
The issue here is that the units can't be strings. If the function ran each element of the set through `Unit`, then it could be the must shorter Unit-like.
I think this might be a good followup PR.

Concurrently, we can add RST substitutions to the file ``conf.py``.

```text

.. RST SUBSTITUTIONS

.. |Unit|  replace:: :class:`~astropy.units.Unit`
.. |Representation|  replace:: :class:`~astropy.coordinates.BaseRepresentation`

```

This nicely compliments the x-refs for terms like Unit which do not (and should not) have glossary definitions. Also, these subs are available in the body of docstrings, not only the parameter definitions, and anywhere in the docs.

Reply:
Coming back to this now: let's postpone to possible follow-up. I do think we don't need the explicit `:class:`, though (see above)

Reply:
✅